modules/ingester/flush.go

```diff
@@ -349,20 +362,15 @@ func (i *Ingester) enqueue(op *flushOp, jitter bool) {
 
 	op.at = time.Now().Add(delay)
 
-	go func() {
-		time.Sleep(delay)
-
-		// Check if shutdown initiated
-		if i.flushQueues.IsStopped() {
-			handleAbandonedOp(op)
-			return
-		}
-
-		err := i.flushQueues.Enqueue(op)
-		if err != nil {
-			handleFailedOp(op, err)
-		}
-	}()
+	if jitter {
```

Comment:
stylistic nit: i would have started the method with:
```
if !jitter {
  i.enqueueExec(op)
  return
}

<rest of func can assume jitter = true>
```

i think it would be clearer, but not wrong the way you've written it